### ISSUE TITLE
GitHub status upload/download should use a combined status instead of a list of all statuses

### ISSUE THREAD

**Issue by tomgeorge** [tektoncd/pipeline#2062]
# Expected Behavior

When using a PullRequest resource, I expect to have the most recent check statuses available to me in a pipeline

# Actual Behavior

A ref can have multiple statuses with the same label,  and `Download`/`toDisk` writes the last occurence of the returned statuses to disk, which is not always the most recent one.  Using `FindCombinedStatus` instead of `ListStatuses` fixes this problem.  A combined status contains the latest version of the checks.


# Steps to Reproduce the Problem

1.  Open a PR and set a check to success/failure back and forth a few times
2.  Run a pipeline that takes a GitHub PR as a resource, and cat out the statuses
```yaml
- name: check-status   
   image: busybox                                                                                                                                                                                                        
   script: |                                                                                                                                                                                                                     
     cat $(outputs.resources.pr.path)/status/*                                                                                                                                                                                   
```
3.  The `state` of the check is not always the most recent one

A full example of the trigger/pipeline is available [here](https://github.com/tomgeorge/plumbing/blob/check-pr-has-kind-label/tekton/resources/checks/check-pr-has-kind-label.yaml)

# Additional Info

**Comment by tomgeorge** [comment]
I started working on this and it works for Github, but basically none of the other git drivers implement `FindCombinedStatus`.  

And you can't implement "use latest status" login in tekton itself without changing the jenkins-x go-scm code.  Tekton only knows about the `scm.Status` struct, which does not have the created at/updated at fields.  Those are in an internal struct in the various git providers code, see https://github.com/jenkins-x/go-scm/blob/cc0bf0ce64091d3c88ca13ff5f5699243ed2f1a2/scm/driver/github/repo.go#L345

**Comment by tomgeorge** [comment]
@vdemeester is this a bug instead of a feature?  You cannot rely on `pullrequest-init` to get the most recent status for each check in a PR.

**Comment by afrittoli** [comment]
I filed tektoncd/pipeline#2188, but it's kind of a duplicate of this one.

**Comment by afrittoli** [comment]
> @vdemeester is this a bug instead of a feature? You cannot rely on `pullrequest-init` to get the most recent status for each check in a PR.

I definitely think this is a bug - see tektoncd/pipeline#2188 also.

**Comment by vdemeester** [comment]
@afrittoli @tomgeorge is this still a bug or has it been fixed (with #2188) ?

**Comment by tomgeorge** [comment]
Sorry for the delay, this has been fixed. I think this can be closed
